services/publicapi/get_transaction.go

```diff
@@ -0,0 +1,84 @@
+package publicapi
+
+import (
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/client"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/pkg/errors"
+)
+
+func (s *service) GetTransactionStatus(input *services.GetTransactionStatusInput) (*services.GetTransactionStatusOutput, error) {
+	if input.ClientRequest == nil {
+		err := errors.Errorf("error: missing input (client request is nil)")
+		s.logger.Info("get transaction status received missing input", log.Error(err))
+		return nil, err
+	}
+
+	s.logger.Info("get transaction status request received", log.String("flow", "checkpoint"), log.Stringable("txHash", input.ClientRequest.Txhash()))
+	txReceipt, err := s.transactionPool.GetCommittedTransactionReceipt(&services.GetCommittedTransactionReceiptInput{
```

Comment:
I'd split this method to two internal calls:
1) getFromTXPool
2) getFromBlockStorage

For the sake of readability and SRP